CRM/Contact/BAO/Query.php

```diff
@@ -1384,8 +1384,7 @@ public function query($count = FALSE, $sortByChar = FALSE, $groupContacts = FALS
       }
     }
     elseif ($sortByChar) {
-      $select = 'SELECT DISTINCT LEFT(contact_a.sort_name, 1) as sort_name';
```

Comment:
So my underlying concern here is that custom searches might be doing the thing that we cleaned up in core & relying on this - I'd rather a deprecation notice than removing these lines

Reply:
Or possibly just a comment if this param is still unavoidably pass in here

Reply:
I've replaced it with a deprecated function warning.  Though I'm not totally convinced about leaving all this stuff lying around in the code and would prefer that any custom searches are just forced to upgrade their code (as they could detect the presence of the alphabetQuery function anyway) - this particular function was responsible for a massive performance issue with smartgroups and I'm not convinced it will actually work "the old way" with the changes we've made anyway.